### PR TITLE
Update kernels (multiple times) and add security related configs

### LINKED ISSUE #2749: [RFC] Enable some security related kernel options
This is a RFC to enable some additional kernel features
- Enable `HAVE_GCC_PLUGINS`. This then allows us to enable:
  - `GCC_PLUGIN_STRUCTLEAK`: Zero init structs which have __user attribute
  - `GCC_PLUGIN_STRUCTLEAK_BYREF_ALL`: Extension of the above
  - `GCC_PLUGIN_RANDSTRUCT`: Randomise the layout of structs
  - `GCC_PLUGIN_RANDSTRUCT_PERFORMANCE`: This is a slightly weaker version of the above but reduces the perf hit by cache aligning the structs.

- Enable `REFCOUNT_FULL`: This performs full refcount validation and has a slight perf hit, which may already been fixed for some arches.

@riyazdf @justincormack @ijc WDYT?

Sources:
- [4.14 security update](https://outflux.net/blog/archives/2017/11/14/security-things-in-linux-v4-14/)
- [4.13 security update](https://outflux.net/blog/archives/2017/09/05/security-things-in-linux-v4-13/)

### PR DESCRIPTION
**Pull request by rn**
- Enable `GCC_PLUGIN_STRUCTLEAK`/`GCC_PLUGIN_RANDSTRUCT`/`REFCOUNT_FULL` (closes #2749)
- Enabling `GCC_PLUGIN` required addition of `mpc1-dev`/`mpfr-dev` to the alpine base and kernel compile images
- Update kernel to 4.13.14/4.9.63/4.4.99
- Update kernel to 4.14.1/4.13.15/4.9.64/4.4.100
- Tweak the content of the `kernel-dev.tar` package to allow building kernel modules with `GCC_PLUGIN_RANDSTRUCT`
- Add tests to build kernel modules for all supported kernels. In particular with `GCC_PLUGIN_RANDSTRUCT` there are some additional complications with compiling external modules so we should test it.
- While at it also fix `Dockerfile.kbuild`, which was broken a while back.

pengu-3

### CODE REVIEW

**Comment by rn** [comment]
`x86_64` kernels are already pushed. Second batch of `arm64` kernels is still building

**Comment by rn** [comment]
CI failed pulling `linuxkit/modprobe:2d153653d1111877a8f53704ef85063a44182ade`. Seems transient and unrelated. Hit rebuild

File: kernel/Makefile [hunk 1]
```diff
@@ -172,13 +172,12 @@ endef
...
-$(eval $(call kernel,4.14,4.14.x,$(EXTRA)))
-$(eval $(call kernel,4.14,4.14.x,-dbg))
-$(eval $(call kernel,4.13.13,4.13.x,$(EXTRA)))
-$(eval $(call kernel,4.13.13,4.13.x,-dbg))
-$(eval $(call kernel,4.9.62,4.9.x,$(EXTRA)))
-$(eval $(call kernel,4.9.62,4.9.x,-dbg))
-$(eval $(call kernel,4.4.98,4.4.x,$(EXTRA)))
+$(eval $(call kernel,4.14.1,4.14.x,$(EXTRA)))
+$(eval $(call kernel,4.14.1,4.14.x,-dbg))
+$(eval $(call kernel,4.13.15,4.13.x,$(EXTRA)))
```
Comment (ijc) [inline kernel/Makefile]: Did you intend to drop the 4.13 dgb kernel?

Comment (rn) [reply kernel/Makefile]: yes as per commit https://github.com/linuxkit/linuxkit/pull/2767/commits/717829ea89b9ad1f10fad3d6c32a1364518557ac

Comment (ijc) [reply kernel/Makefile]: thanks, I only looked at the PR description!

Comment (rn) [reply kernel/Makefile]: yeah, I should have mentioned it there but forgot...too many changes

**Pull request merged**